src/generators/java/presets/ConstraintsPreset.ts

```diff
@@ -0,0 +1,57 @@
+import { JavaPreset } from '../JavaPreset';
+
+import { CommonModel } from '../../../models';
+
+/**
+ * Preset which extends class's getters with annotations from `javax.validation.constraints` package
+ * 
+ * @implements {JavaPreset}
+ */
+export const JAVA_CONSTRAINTS_PRESET: JavaPreset = {
+  class: {
+    getter({ renderer, model, propertyName, property, content }) {
+      if (!(property instanceof CommonModel)) {
+        return content;
+      }
+      const annotations: string[] = [];
+      
+      const isRequired = model.isRequired(propertyName);
+      if (isRequired) {
+        annotations.push(renderer.renderAnnotation('NotNull'));
+      }
+    
+      const pattern = property.getFromSchema('pattern');
+      if (pattern !== undefined) {
+        annotations.push(renderer.renderAnnotation('Pattern', { regexp: `"${pattern}"` }));
+      }
+    
+      const minimum = property.getFromSchema('minimum');
+      if (minimum !== undefined) {
+        annotations.push(renderer.renderAnnotation('Min', minimum));
+      }
+    
+      const exclusiveMinimum = property.getFromSchema('exclusiveMinimum');
+      if (exclusiveMinimum !== undefined) {
+        annotations.push(renderer.renderAnnotation('Min', exclusiveMinimum + 1));
+      }
+    
+      const maximum = property.getFromSchema('maximum');
+      if (maximum !== undefined) {
+        annotations.push(renderer.renderAnnotation('Max', maximum));
+      }
+    
+      const exclusiveMaximum = property.getFromSchema('exclusiveMaximum');
+      if (exclusiveMaximum !== undefined) {
+        annotations.push(renderer.renderAnnotation('Max', exclusiveMaximum - 1));
+      }
+    
+      const minItems = property.getFromSchema('minItems');
+      const maxItems = property.getFromSchema('maxItems');
+      if (minItems !== undefined || maxItems !== undefined) {
+        annotations.push(renderer.renderAnnotation('Size', { min: minItems, max: maxItems }));
+      }
```

Comment:
I feel like I have to leave a comment here 😄 In complicated JSON Schemas (when keywords for combination or conditional are used), this will not give the desired output. 

I know you added it because the java-spring-template had them. But they should be simplified, and even that process is difficult and needs to be well thought out how you merge such formatting rules.

IMO rendering these formatting rules when they have not been simplified will in the worse cases not give a useable output for the user. In the common cases yes it will work, but if there is just one combination or conditional keyword described contradicting formatting rules this will not provide a useable outcome.
i.e. 
```
{
  "type": "string",
  "maxLength": 50,
  "allOf": [
    { "maxLength": 2 }
    { "maxLength": 5 }
  ]
}
```
Should only work with strings between 0 and 5 characters but with this implementation up to 50 is allowed.

Reply:
Yes, I know about this problem, but as you wrote in main comment, I will create followup issue for tracking such a cases. For "simple" metadata fields, not related to any type, like `description` etc `getFromSchema` function should work without any problems, but with combined schema and types like `number`, `string` etc we should simplify it. At the moment any template doesn't support nested metadata, so right now it's not a big problem.

Reply:
Here is a issue https://github.com/asyncapi/generator-model-sdk/issues/130